stream_alert/rule_processor/handler.py

```diff
@@ -167,68 +171,141 @@ def get_alerts(self):
         """
         return self._alerts
 
+    @staticmethod
+    def _segment_records_by_count(record_list, max_count):
+        """Segment records by length
+
+        Args:
+            record_list (list): The original records list to be segmented
+            max_count (int): The max amount of records to yield per group
+        """
+        for item in range(0, len(record_list), max_count):
+            yield record_list[item:item + max_count]
+
+    def _segment_records_by_size(self, record_batch):
+        """Segment record groups by size
+
+        Args:
+            record_batch (list): The original record batch to measure and segment
+
+        Returns:
+            generator: Used to iterate on each newly segmented group
+        """
+        split_factor = 1
+        len_batch = len(record_batch)
+
+        while True:
+            if len(str(record_batch[:len_batch / split_factor])) > MAX_BATCH_SIZE:
+                split_factor += 1
+            else:
+                break
```

Comment:
More simply:

```python
# Calculate the appropriate split factor
while len(str(record_batch[:len_batch / split_factor])) > MAX_BATCH_SIZE:
    split_factor += 1
```

Reply:
This also only checks the size of the first batch and assumes every other chunk will be about the same size. Since it's possible that different chunks may be different sizes, can you add a comment explaining why you feel that possibility is unlikely?

Reply:
Sure, good point!